modules/navigation/nav_map.cpp

```diff
@@ -1413,6 +1414,40 @@ void NavMap::_update_merge_rasterizer_cell_dimensions() {
 	merge_rasterizer_cell_height = cell_height * merge_rasterizer_cell_scale;
 }
 
+int NavMap::get_region_connections_count(NavRegion *p_region) const {
+	ERR_FAIL_NULL_V(p_region, 0);
+
+	HashMap<NavRegion *, LocalVector<gd::Edge::Connection>>::ConstIterator E = region_external_connections.find(p_region);
+	if (E) {
+		return E->value.size();
+	}
+	return 0;
+}
+
+Vector3 NavMap::get_region_connection_pathway_start(NavRegion *p_region, int p_connection_id) const {
+	ERR_FAIL_NULL_V(p_region, Vector3());
+
+	HashMap<NavRegion *, LocalVector<gd::Edge::Connection>>::ConstIterator E = region_external_connections.find(p_region);
+	if (E) {
+		ERR_FAIL_INDEX_V(p_connection_id, int(E->value.size()), Vector3());
+		return E->value[p_connection_id].pathway_start;
+	}
+
+	return Vector3();
+}
+
+Vector3 NavMap::get_region_connection_pathway_end(NavRegion *p_region, int p_connection_id) const {
+	ERR_FAIL_NULL_V(p_region, Vector3());
+
+	HashMap<NavRegion *, LocalVector<gd::Edge::Connection>>::ConstIterator E = region_external_connections.find(p_region);
+	if (E) {
+		ERR_FAIL_INDEX_V(p_connection_id, int(E->value.size()), Vector3());
+		return E->value[p_connection_id].pathway_end;
```

Comment:
Let's rename `E` to something more meaningful like `lookup_result` `find_result` or smth else.